platform/swift/source/LoggerBridge.swift

```diff
@@ -30,6 +55,14 @@ final class LoggerBridge: LoggerBridging {
         network: Network?,
         errorReporting: RemoteErrorReporting
     ) {
+        do {
+            try bufferDirectoryPath.map(makeDirectoryAndDisableProtection(at:))
+        } catch {
+            // To be safe we don't initialize the logger if we can't create the directory or set
+            // the file protection policy.
+            return nil
+        }
+
         let loggerID = capture_create_logger(
```

Comment:
I don't think so but just checking that the platform code doesn't make any files before we get here? I know there is https://github.com/bitdriftlabs/capture-sdk/blob/main/platform/swift/source/src/key_value_storage.rs but this is used by the rust code only AFAIK to store things like device ID, etc. On that note, does whatever that use need any permission changes or it's a separate thing?

Reply:
separate thing. I think we should be good with userdefaults. will double check though